api/handlers.go

```diff
@@ -449,6 +449,9 @@ func SearchHandler(c *gin.Context) {
 	if _, ok := consts.SORT_BY_VALUES[sortBy]; ok {
 		sortByVal = sortBy
 	}
+	if len(query.Term) == 0 {
```

Comment:
I need a big fat comment before this line explaining why this is needed and why it covers all cases properly.
This is also not true, because if someone looks for exact match?
You need to add this to the condition.

It feels too breakable. Need to think of something that we will not forget to update if needed.
Maybe add a function to Query struct that answers a boolean if the QueryIsFilterOnly

Also in the comment add reference to the specific bug (link to bug).